src/chapters/control-flow/loops.rst

```diff
@@ -0,0 +1,278 @@
+----------------------------
+
+The **continue** statement is similar to, but importantly different
+from, the ``break`` statement. Like ``break``, it interrupts the normal
+flow of control of the loop. But unlike ``break``, the ``continue``
+statement only terminates the *current iteration* of the loop. So the
+loop will continue to run from the top (as long as the boolean
+expression that controls the loop is still true) after a ``continue``
+statement. Here is an example:
+
+.. sourcecode:: java
+   :linenos:
+
+   public class testContinue {
+
+      public static void main(String [] args) {
+         int[] someInts = {1, 10, 2, 3, 5, 8, 10};
+         int searchTerm = 10;
+         for (int oneInt : someInts) {
+            if (oneInt == searchTerm) {
+               System.out.println("Found it!");
+               continue;
+            }
+            System.out.println("Not here");
+         }
+      }
+   }
+
+The above program will print "Not here" on every iteration of the
+``for`` loop *except* where the number has been found. So the output
+looks like this:
+
+.. sourcecode:: bash
+
+   Not here
+   Found it!
+   Not here
+   Not here
+   Not here
+   Not here
+   Found it!
+
+Because of the ``continue`` statement, the final print statement in the
+for loop is skipped. If the ``continue`` statement weren’t there, the
+output would look like this instead (notice the extra "Not here"
+printouts):
+
+.. sourcecode:: bash
+
+   Not here
+   Found it!
+   Not here
+   Not here
+   Not here
+   Not here
+   Not here
+   Found it!
+   Not here
+
+References
+----------
+
+-  `The for statement
+   (docs.oracle.com) <https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html>`__
+-  `The while and do-while Statements
+   (docs.oracle.com) <https://docs.oracle.com/javase/tutorial/java/nutsandbolts/while.html>`__
+-  `Break and Continue Statements
+   (docs.oracle.com) <https://docs.oracle.com/javase/tutorial/java/nutsandbolts/branch.html>`__
+-  `Summary of Control Flow Statements
+   (docs.oracle.com) <https://docs.oracle.com/javase/tutorial/java/nutsandbolts/flowsummary.html>`__
+
+Check Your Understanding
+-------------------------
+
+.. admonition:: Question
+
+   .. sourcecode:: java
+      :linenos:
+
+      char chars[] = {'p', 'l', 'r', 's', 't'};
+
+      for (<conditional>) {
+         System.out.println(i);
+      }
+
+   What does the missing <conditional> statement need to be to print each item in ``chars``?
+
+   #. char i : chars
+   #. char i : chars[]
+   #. char i in chars
+   #. char i in chars[]
+
+.. admonition:: Question
+
+   .. sourcecode:: java
+      :linenos:
+
+      do {
+         System.out.println("Hello world!");
+      } while (3 < 2);
+
+   How many times does the message print and why?
+
+   #. 0 - The ``while`` condition is never true.
+   #. 1 - The print statement is evaluated before the conditional.
+   #. infinite times - ``do-while`` loops always run forever.
+   #. 1 - 3 is less than 2.
```

Comment:
I think options c and d can be combined to avoid hyperbole.  Maybe something like, "infinite times---3 is less than 2, and the condition is never changed in the loop". Then you can add another choice option.

Reply:
i don't find these options to be hyperbole. Do you have another option in mind?

Reply:
The issue is with the "do-while loops always run forever".  Since we try to avoid infinite loops, it would be silly for Oracle to purposefully introduce a tool for creating one.  That's why I suggested an edit for option "c".

Reply:
I prefer to keep the options as is if its a personal preference. we often give the students silly/unreasonable options in these types of questions and I like having 4 options over 3, somewhat for aesthetics

Reply:
I think we can revise part "c" and add another option.  However, having a non-nonsensical choice is just the same as presenting fewer options.